### PR TITLE
feat: Configure Orleans for running on Azure

### PR DESCRIPTION
**Pull request by akselkvitberg**
Configures the server to use localhost clustering and in memory stores when running locally and Azure Storage when running in production.

### CODE REVIEW

**Review by rvanoord — APPROVED** [review]

File: src/Api/Orleans.cs [hunk 1]
```diff
@@ -6,12 +11,56 @@ public static void AddOrleans(this WebApplicationBuilder builder)
...
+                // Is running as an App Service?
+                if (builder.Configuration["WEBSITE_PRIVATE_IP"] is { } ip &&
+                    builder.Configuration["WEBSITE_PRIVATE_PORTS"] is { } ports)
+                {
+                    var endpointAddress = IPAddress.Parse(ip);
+                    var splitPorts = ports.Split(',');
+                    if (splitPorts.Length < 2)
+                        throw new Exception("Insufficient private ports configured.");
+
+                    siloPort = int.Parse(splitPorts[0]);
```
Comment (runenilsenoe) [inline src/Api/Orleans.cs:38]: this could probably be easier to split into separate envs. more explicit

File: src/Api/Orleans.cs [hunk 2]
```diff
@@ -6,12 +11,56 @@ public static void AddOrleans(this WebApplicationBuilder builder)
...
+                    silo.ConfigureEndpoints(endpointAddress, siloPort, gatewayPort);
+                }
+                else // Assume Azure Container Apps.
+                {
+                    silo.ConfigureEndpoints(siloPort, gatewayPort);
+                }
+                
+                silo.Configure<ClusterOptions>(options =>
+                {
+                    options.ClusterId = "webhookCluster";
```
Comment (runenilsenoe) [inline src/Api/Orleans.cs:50]: the name webhook doesnt make much sense to me here, but ok

File: src/Api/Orleans.cs [hunk 3]
```diff
@@ -6,12 +11,56 @@ public static void AddOrleans(this WebApplicationBuilder builder)
...
-                .AddReminders()
-                .AddMemoryGrainStorage("store")
-                .UseDashboard();
+            silo.UseDashboard();
+            
+            if (builder.Environment.IsDevelopment())
+            {
+                silo.UseLocalhostClustering()
+                    .AddMemoryGrainStorageAsDefault()
+                    .ConfigureEndpoints("localhost", 9889, 9099, AddressFamily.InterNetwork, true)
```
Comment (runenilsenoe) [inline src/Api/Orleans.cs:20]: specifying ports is enough here. easier read 
`.ConfigureEndpoints(9889,9099)`

File: src/Grains/OrleansConstants.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace Grains;
+
+public class OrleansConstants
+{
+    public const string StorageName = "store";
```
Comment (runenilsenoe) [inline src/Grains/OrleansConstants.cs:5]: this need to be added to the grains and to the grain storage specification in silo build. But good idea

**Review by runenilsenoe — COMMENTED** [review]
Nice

File: src/Api/Orleans.cs (hunk 1 above)
Comment (akselkvitberg) [reply src/Api/Orleans.cs:38]: It's defined by Azure by setting `vnetPrivatePortsCount: 2` when creating the azure app service.
https://learn.microsoft.com/en-us/dotnet/orleans/deployment/deploy-to-azure-app-service

File: src/Api/Orleans.cs (hunk 2 above)
Comment (akselkvitberg) [reply src/Api/Orleans.cs:50]: pubsub is probably better

**Pull request merged**